.github/workflows/unit_tests_with_featuretools_main_branch.yml

```diff
@@ -0,0 +1,54 @@
+name: Unit Tests - EvalML with Featuretools main branch
+on:
+  workflow_dispatch:
+jobs:
+  unit_tests_featuretools_main:
+    if: ${{ github.repository_owner == 'alteryx' }}
+    name: ${{ matrix.python_version }} unit tests ${{ matrix.libraries }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python_version: ["3.7", "3.8", "3.9", "3.10"]
+        libraries: ["core"]
+    steps:
+      - name: Set up python ${{ matrix.python_version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python_version }}
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Build featuretools package
+        run: make package_featuretools
+      - name: Set up pip and graphviz
+        run: |
+          pip config --site set global.progress_bar off
+          python -m pip install -U pip
+          sudo apt update && sudo apt install -y graphviz
+      - name: Install EvalML & Featuretools - test requirements
+        run: |
+          python -m pip install -e ./
+          python -m pip uninstall -y featuretools
+          python -m pip install https://github.com/alteryx/featuretools/archive/main.zip
+      - name: Run unit tests without code coverage
+        run: |
+          echo "Run unit tests without code coverage for ${{ matrix.python_version }} and ${{ matrix.libraries }}"
+          echo "Testing with EvalML version:" `python -c "import evalml; print(evalml.__version__)"`
+          pytest evalml/ -n 2
+  slack_alert_failure:
+    name: Send Slack alert if failure
+    needs: unit_tests_featuretools_main
+    runs-on: ubuntu-latest
+    if: ${{ always() }}
+    steps:
+      - name: Send Slack alert if failure
+        if: ${{ needs.unit_tests_featuretools_main.result != 'success' }}
+        id: slack
+        uses: slackapi/slack-github-action@v1
+        with:
+          payload: |
+            {
+              "url": "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+            }
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
```

Comment:
You will need to setup a Slack webhook
- https://github.com/slackapi/slack-github-action#setup
I added you as a collab on the Featuretools one so you can see what it looks like